internal/commands/app/create.go

```diff
@@ -109,56 +112,60 @@ func (cmd *CommandCreate) Handler(profile *cli.Profile, ui terminal.UI) error {
 		}
 	}
 
-	loadedApp, loadedAppErr := local.LoadApp(dir)
-	if loadedAppErr != nil {
-		return loadedAppErr
+	loadedApp, err := local.LoadApp(dir)
+	if err != nil {
+		return err
 	}
 
-	newApp, newAppErr := cmd.realmClient.CreateApp(groupID, cmd.inputs.Name, realm.AppMeta{cmd.inputs.Location, cmd.inputs.DeploymentModel})
-	if newAppErr != nil {
-		return newAppErr
+	newApp, err := cmd.realmClient.CreateApp(groupID, cmd.inputs.Name, realm.AppMeta{cmd.inputs.Location, cmd.inputs.DeploymentModel})
+	if err != nil {
+		return err
 	}
 
-	// TODO(REALMC-8135): Implement data-source flag for app create command
-	// dsCluster, dsClusterErr := cmd.inputs.resolveDataSource(cmd.realmClient, cmd.inputs.Project, app.ID())
-	// if dsClusterErr != nil {
-	// 	return dsClusterErr
-	// }
-	// service := realm.ServiceDescData{
-	// 	Config: map[string]interface{}{
-	// 		"clusterName":         dsCluster,
-	// 		"readPreference":      "primary",
-	// 		"wireProtocolEnabled": false,
-	// 	},
-	// 	Name: appName + "_cluster",
-	// 	Type: "mongodb-atlas",
-	// }
-	// Potentially try to use Import to create app Service
-	// _, dsErr := cmd.realmClient.CreateAppService(cmd.inputs.Project, app.ID(), service) // for output possibly
-	// if dsErr != nil {
-	// 	return dsErr
-	// }
-	// Won't need if import works
-	// zipPkgName, zipPkg, exportErr := cmd.realmClient.Export(
-	// 	cmd.inputs.Project,
-	// 	app.ID(),
-	// 	realm.ExportRequest{ConfigVersion: realm.DefaultAppConfigVersion},
-	// )
-	// if exportErr != nil {
-	// 	return exportErr
-	// }
-	// if idx := strings.LastIndex(zipPkgName, "_"); idx != -1 {
-	// 	zipPkgName = zipPkgName[:idx]
-	// }
-	// zipPkgPath := filepath.Join(dir, zipPkgName) // Is this the correct path
-	// if writeErr := local.WriteZip(zipPkgPath, zipPkg); writeErr != nil {
-	// 	return writeErr
-	// }
+	if cmd.inputs.DataSource != "" {
+		dsCluster, err := cmd.inputs.resolveDataSource(cmd.realmClient, cmd.inputs.Project, newApp.ID)
+		if err != nil {
+			return err
+		}
+		serviceName := newApp.Name + "_cluster"
+		serviceConfig := map[string]interface{}{
+			"name": serviceName,
+			"type": "mongodb-atlas",
+			"config": map[string]interface{}{
+				"clusterName":         dsCluster,
+				"readPreference":      "primary",
+				"wireProtocolEnabled": false,
+			},
```

Comment:
[major q] Should we consider data lakes here as well? Data Lakes, support for which is relatively new (mid-late 2020), and Clusters are the two big types of data sources we have. I believe the service configuration is different as well for data lakes so that is something to keep in mind. Though I haven't looked through the master branch yet to see if we added support for data lakes in the cli last year

Reply:
this flag is new functionality, so there was no prior support for this.  this is a great point though... lets tackle that in a separate jira though